### PR TITLE
Makefile: add uninstall target

### LINKED ISSUE #30: Provide a make uninstall
Well, in case we want to remove all installed stuff 👯

### PR DESCRIPTION
**Pull request by erikw**
Fixes #30

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 SRCS_SCRIPTS	= $(filter-out %cron_mail, $(wildcard usr/local/sbin/*))
-SRCS_CONF	= $(filter-out %template, $(wildcard etc/restic/*))
+SRCS_CONF		= $(filter-out %template, $(wildcard etc/restic/*))
 SRCS_SYSTEMD	= $(wildcard etc/systemd/system/*)
 
 # Just set PREFIX in envionment, like
 # $ PREFIX=/tmp/test make
-DEST_SCRIPTS	= $(PREFIX)/usr/local/sbin
-DEST_CONF	= $(PREFIX)/etc/restic
+DEST_SCRIPT		= $(PREFIX)/usr/local/sbin
```
Comment (gerardbosch) [inline Makefile:11]: Hi @erikw I've just seen something strange here (I have very little knowledge about make, but) I see you renamed to singular: `DEST_SCRIPT` but, the references below at line 33 and 34 did not get updated accordingly.

I'm guessing this is gonna be a problem, isn't it?

File: Makefile [hunk 2]
```diff
@@ -46,3 +49,11 @@ install-conf: | etc/restic/b2_env.sh etc/restic/b2_pw.txt
 install-systemd:
 	install -d $(DEST_SYSTEMD)
 	install -m 0644 $(SRCS_SYSTEMD) $(DEST_SYSTEMD)
+
+			#$(RM) $$file; \
```
Comment (gerardbosch) [inline Makefile:53]: Is this intended? Looked like a dead comment :upside_down_face: I'm not very sure.

File: Makefile (hunk 1 above)
Comment (erikw) [reply Makefile:11]: Thanks! Fixed in 88f2dc9eca0400f94ff3e84c5de1067dfca694a4

File: Makefile (hunk 2 above)
Comment (erikw) [reply Makefile:53]: Nope, you're completely right. :D 

16e3cb2df58186cbe004b16c8a1973c702cbdc9d

**Comment by gerardbosch** [comment]
Thanks @erikw I updated the README to include the uninstall instructions.
I opened a very small PR: https://github.com/erikw/restic-systemd-automatic-backup/pull/39
You're free to edit it directly if you want :)

**Comment by gerardbosch** [comment]
_Disclaimer: Maybe I'm understanding the following wrong as don't have any background with make_

One more question @erikw, will the `make uninstall` work if `PREFIX=<some-path> make uninstall` is specified?

I see this:

```bash
# Just set PREFIX in envionment, like
# $ PREFIX=/tmp/test make
DEST_SCRIPTS	= $(PREFIX)/usr/local/sbin
DEST_CONF	= $(PREFIX)/etc/restic
DEST_SYSTEMD	= $(PREFIX)/etc/systemd/system

INSTALLED_FILES = $(patsubst %, /%, $(SRCS_SCRIPTS)) \
		  $(patsubst %, /%, $(SRCS_CONF)) $(DEST_CONF)/b2_env.sh $(DEST_CONF)/b2_pw.txt \
		  $(patsubst %, /%, $(SRCS_SYSTEMD))
```

Which seems that PREFIX could be used to test installation it in a tmp dir for example.
But from my little understanding I see this will not apply to uninstall, as prefix is not in the substitution. I'm not very sure if it should or would make sense anyway. Just wanted to find symmetry b/w install/uninstall.

Would this make sense or even work?

```bash
INSTALLED_FILES = $(patsubst %, $(PREFIX)/%, $(SRCS_SCRIPTS)) \   # notice the PREFIX here
# ...
```

**Comment by erikw** [comment]
You are very correct Sir @gerardbosch !

I think 8f43977b3e79e0f24e097dea728589a3a49ebab4 should take care of it:

```console
$ make uninstall
rm -f /usr/local/sbin/nm-unmetered-connection.sh
rm -f /usr/local/sbin/restic_backup.sh
rm -f /usr/local/sbin/restic_check.sh
rm -f /usr/local/sbin/systemd-email
rm -f /etc/restic/b2_env.sh
rm -f /etc/restic/b2_pw.txt
rm -f /etc/restic/backup_exclude
rm -f /etc/systemd/system/nm-unmetered-connection.service
rm -f /etc/systemd/system/restic-backup.service
rm -f /etc/systemd/system/restic-backup.timer
rm -f /etc/systemd/system/restic-check.service
rm -f /etc/systemd/system/restic-check.timer
rm -f /etc/systemd/system/status-email-user@.service
rm -f /etc/restic/b2_env.sh
rm -f /etc/restic/b2_pw.txt
$ PREFIX=/tmp make uninstall
rm -f /tmp/usr/local/sbin/nm-unmetered-connection.sh
rm -f /tmp/usr/local/sbin/restic_backup.sh
rm -f /tmp/usr/local/sbin/restic_check.sh
rm -f /tmp/usr/local/sbin/systemd-email
rm -f /tmp/etc/restic/b2_env.sh
rm -f /tmp/etc/restic/b2_pw.txt
rm -f /tmp/etc/restic/backup_exclude
rm -f /tmp/etc/systemd/system/nm-unmetered-connection.service
rm -f /tmp/etc/systemd/system/restic-backup.service
rm -f /tmp/etc/systemd/system/restic-backup.timer
rm -f /tmp/etc/systemd/system/restic-check.service
rm -f /tmp/etc/systemd/system/restic-check.timer
rm -f /tmp/etc/systemd/system/status-email-user@.service
rm -f /tmp/etc/restic/b2_env.sh
rm -f /tmp/etc/restic/b2_pw.txt
```

**Comment by gerardbosch** [comment]
Nice!! Great to see this automation tool improving :tada:

**Pull request merged**